examples/step-37/step-37.cc

```diff
@@ -871,7 +872,7 @@ namespace Step37
         IndexSet relevant_dofs;
         DoFTools::extract_locally_relevant_level_dofs(
           dof_handler, level, relevant_dofs);
-        ConstraintMatrix level_constraints;
+        AffineConstraints<float> level_constraints;
```

Comment:
... we now need a `AffineConstraints<float>` object for the level constraints (when you use a smoother operating on floating point numbers).

Reply:
This is also somewhat nasty, but we might get around this at a later time by allowing a different type of constraints as the `Number` type, as long as the argument here can be converted into `Number`.